File: app/src/main/java/com/rdd/finy/repository/WalletRepository.java
package com.rdd.finy.repository;

import android.arch.lifecycle.LiveData;
import com.rdd.finy.data.Wallet;

import java.util.List;

public interface WalletRepository {

    LiveData<Wallet> findById(int id);

    LiveData<List<Wallet>> findAll();

    long insert(Wallet wallet);

    int delete(Wallet wallet);

}

File: app/src/main/java/com/rdd/finy/repository/WalletRepository.kt
package com.rdd.finy.repository

import android.arch.lifecycle.LiveData
import com.rdd.finy.data.Wallet


interface WalletRepository {
    fun findById(id: Int): LiveData<Wallet>

    fun findAll(): LiveData<List<Wallet>>

    fun insert(wallet: Wallet): Long

    fun delete(wallet: Wallet): Int

}